src/app/issues/page.tsx

```diff
@@ -0,0 +1,330 @@
+"use client";
+
+import * as React from "react";
+
+import { clsx } from "clsx";
+import {
+  Check,
+  CheckCircle2,
+  ChevronDown,
+  CircleDot,
+  GitMerge,
+  MessageSquare,
+  Search,
+} from "lucide-react";
+
+export default function IssuesPage({}) {
+  const [issueType, setIssueType] = React.useState<
+    "created" | "assigned" | "mentioned"
+  >("created");
+
+  const [search, setSearch] = React.useState<string>(
+    `is:open is:issue author:xxxxx archived:false`
+  );
+  const [issues, setIssues] = React.useState<IIssueData[]>(openData);
+  const [issueStatus, setIssueStatus] = React.useState<"open" | "closed">(
+    "open"
+  );
+
+  React.useEffect(() => {
+    if (issueStatus === "open") {
+      setIssues(openData);
+    } else {
+      setIssues(closedData);
+    }
+  }, [issueStatus]);
+
+  return (
+    <>
+      <section className="sm:px-8 py-6 max-w-6xl m-auto">
+        <div className="md:flex justify-between gap-4 px-4 sm:px-0">
+          <div className="">
+            <span className="isolate inline-flex rounded-md shadow-sm w-full">
+              <button
+                type="button"
+                className={clsx(
+                  "relative inline-flex w-full md:w-min items-center rounded-l-md px-4 py-1.5 font-semibold text-slate-300 ring-1 ring-inset ring-slate-600 focus:z-10",
+                  {
+                    "bg-purple-600 text-slate-50": issueType === `created`,
+                  }
+                )}
+                onClick={() => setIssueType("created")}

```

Comment:
make a callback before the return

Reply:
@pierresisson please elaborate or give an example